pkg/controller/applicationmonitoring/templateHelper.go

```diff
@@ -234,3 +240,21 @@ func GeneratePassword(n int) (string, error) {
 
 	return base64.StdEncoding.EncodeToString(b), err
 }
+
+// PopulateAffinityRule returns the yaml representation of the affinity scheduling rule
+func PopulateAffinityRule(affinity *corev1.Affinity) string {
+	// Check if the affinty rule is empty
+	if affinity.Size() < 1 {
+		return ""
+	}
+
+	// Marshall affinty rule into yaml format
+	affinityRule, err := yaml.Marshal(affinity)
+	if err != nil {
+		log.Info(fmt.Sprintf("Error executing PopulateAffinityRule: %s", err))
+		return ""
+	}
+
+	// Indent 2 spaces and return
+	return indent.String("  ", string(affinityRule))
```

Comment:
```suggestion
	return affinityRule 
```

I'd be partial to add the indentation in the template. We can use the [`sprig`](https://github.com/Masterminds/sprig) library that includes the `indent` function (see suggestions below)

The functions can be included by adding the sprig FuncMap to the template object:

https://github.com/integr8ly/application-monitoring-operator/blob/master/pkg/controller/applicationmonitoring/templateHelper.go#L208-L211

Reply:
Great suggestion. I think it's also probably best practice to delegate the formatting/indentation to the template itself, not the code.